src/Integration.Vsix/SonarLintDaemon/IAnalyzerController.cs

```diff
@@ -19,98 +19,23 @@
  */
 
 using System.Collections.Generic;
-using System.ComponentModel.Composition;
-using System.Linq;
 using EnvDTE;
-using SonarLint.VisualStudio.Integration.Vsix.CFamily;
+
+// TODO: decide whether both of these interfaces are required
 
 namespace SonarLint.VisualStudio.Integration.Vsix
 {
-    internal interface IAnalyzerController
+    public interface IAnalyzer
     {
         bool IsAnalysisSupported(IEnumerable<SonarLanguage> languages);
 
         void RequestAnalysis(string path, string charset, IEnumerable<SonarLanguage> detectedLanguages, IIssueConsumer consumer, ProjectItem projectItem);
     }
 
-    [Export(typeof(IAnalyzerController))]
-    internal class AnalyzerController : IAnalyzerController
+    // Marker interface used by for MEF exporting/importing
+    // (there are multiple implementations of IAnalyzer but only one implentation
+    // of this interface).
+    internal interface IAnalyzerController : IAnalyzer
     {
-        private readonly ISonarLintSettings settings;
-        private readonly ISonarLintDaemon daemon;
-        private readonly ILogger logger;
-
-        [ImportingConstructor]
-        public AnalyzerController(
-            ISonarLintSettings settings,
-            ISonarLintDaemon daemon,
-            ILogger logger
-            )
-        {
-            this.settings = settings;
-            this.daemon = daemon;
-            this.logger = logger;
-        }
-
-        public bool IsAnalysisSupported(IEnumerable<SonarLanguage> languages)
-        {
-            if (languages.Contains(SonarLanguage.CFamily))
```

Comment:
Encapsulation: the analyzer controller shouldn't have to know e.g. what languages are supported by the daemon/CLang sub process, and it shouldn't have to know how to invoke them.
The `IAnalyzer` interface provides hides these details from the controller.